pkg/relayer/relayminer_test.go

```diff
@@ -57,3 +60,69 @@ func TestRelayMiner_StartAndStop(t *testing.T) {
 	err = relayminer.Stop(ctx)
 	require.NoError(t, err)
 }
+
+func TestRelayMiner_Ping(t *testing.T) {
+	srObs, _ := channel.NewObservable[*servicetypes.Relay]()
+	servedRelaysObs := relayer.RelaysObservable(srObs)
+
+	mrObs, _ := channel.NewObservable[*relayer.MinedRelay]()
+	minedRelaysObs := relayer.MinedRelaysObservable(mrObs)
+
+	ctx := polyzero.NewLogger().WithContext(context.Background())
+	relayerProxyMock := testrelayer.NewMockOneTimeRelayerProxyWithPing(
+		ctx, t,
+		servedRelaysObs,
+	)
+
+	minerMock := testrelayer.NewMockOneTimeMiner(
+		ctx, t,
+		servedRelaysObs,
+		minedRelaysObs,
+	)
+
+	relayerSessionsManagerMock := testrelayer.NewMockOneTimeRelayerSessionsManager(
+		ctx, t,
+		minedRelaysObs,
+	)
+
+	deps := depinject.Supply(
+		relayerProxyMock,
+		minerMock,
+		relayerSessionsManagerMock,
+	)
+
+	relayminer, err := relayer.NewRelayMiner(ctx, deps)
+	require.NoError(t, err)
+	require.NotNil(t, relayminer)
+
+	err = relayminer.Start(ctx)
+	require.NoError(t, err)
+
+	time.Sleep(time.Millisecond)
+
+	filename := "/tmp/relayerminer.ping.sock"
+
+	ln, err := net.Listen("unix", filename)
+	require.NoError(t, err)
+	defer os.Remove(filename)
+
+	err = relayminer.ServePing(ctx, ln)
+	require.NoError(t, err)
+
+	time.Sleep(time.Millisecond)
+
+	c := http.Client{
+		Transport: &http.Transport{
+			DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
+				return net.Dial(ln.Addr().Network(), ln.Addr().String())
+			},
+		},
```

Comment:
1. Can we please extract the `*http.Transport` value (i.e. keep the `&` here) to a `var`?
2. Are we sure that it's necessary to provide a `DialContext` func? If so, please add a comment explaining.

Reply:
> Are we sure that it's necessary to provide a DialContext func? If so, please add a comment explaining.

Yes, this is required to instantiated an unencrypted TCP connection.

Reply:
Digging in: when I uncomment the http client transport configuration I get the following error:

```
    relayminer_test.go:125: 
        	Error Trace:	/home/bwhite/Projects/pokt/poktroll/pkg/relayer/relayminer_test.go:125
        	Error:      	Received unexpected error:
        	            	Get "http://unix": dial tcp: lookup unix: no such host
        	Test:       	TestRelayMiner_Ping
```

Which makes sense given that our listener is a unix socket. :sweat_smile: 

```suggestion
		// Override transport configuration to adapt the http client to the unix socker listener.
		Transport: &http.Transport{
			DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
				return net.Dial(ln.Addr().Network(), ln.Addr().String())
			},
		},
```